doc/guides/build/index.rst

```diff
@@ -0,0 +1,132 @@
+.. _build_overview:
```

Comment:
I rather prefer make a commit with this content and you move it according you think is proper and review there as well.

Reply:
I'll address all comments and update this pr, ok ?

Reply:
>  As such, it would be a great addition to the overview there ([docs.zephyrproject.org/latest/application/index.html#overview](https://docs.zephyrproject.org/latest/application/index.html#overview)) rather than a separate document.

I disagree, this is NOT about how to develop or build an application, this is a reference of how the build system and build process works, just look at all the figures being added, how is this really going to help someone trying to write a simple application?